[Bug] (be) Possible access of NULL pointer (null_map) at vorc_reader.cpp

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.


### Version

master

### What's Wrong?

doris/be/src/vec/exec/format/orc/vorc_reader.cpp:2173:27: error: Null pointer dereference: null_map [ctunullpointer]
    auto* null_map_data = null_map->data();
doris/be/src/vec/exec/format/orc/vorc_reader.cpp:2173:27: note: Dereferencing argument null_map that is null

Cppcheck report NULL pointer access in vorc_reader

### What You Expected?

Pointer access should be done after validating pointer for NULL check

### How to Reproduce?

cppcheck on be code. and search error 

vallish@Vallishs-MacBook-Air workspace % cat te.txt| grep ctunullpointer
doris/be/src/vec/exec/format/orc/vorc_reader.cpp:2173:27: error: Null pointer dereference: null_map [ctunullpointer]
vallish@Vallishs-MacBook-Air workspace % 


### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/be/src/vec/exec/format/orc/vorc_reader.cpp
+++ b/be/src/vec/exec/format/orc/vorc_reader.cpp
@@ -2170,10 +2170,10 @@ MutableColumnPtr OrcReader::_convert_dict_column_to_string_column(
     const int* dict_data = dict_column->get_data().data();
     string_values.reserve(num_values);
     size_t max_value_length = 0;
-    auto* null_map_data = null_map->data();
     if (orc_column_type->getKind() == orc::TypeKind::CHAR) {
         // Possibly there are some zero padding characters in CHAR type, we have to strip them off.
         if (null_map) {
+            auto* null_map_data = null_map->data();
             for (int i = 0; i < num_values; ++i) {
                 if (!null_map_data[i]) {
                     char* val_ptr;
@@ -2207,6 +2207,7 @@ MutableColumnPtr OrcReader::_convert_dict_column_to_string_column(
         }
     } else {
         if (null_map) {
+            auto* null_map_data = null_map->data();
             for (int i = 0; i < num_values; ++i) {
                 if (!null_map_data[i]) {
                     char* val_ptr;

